mitodl/ocw-to-hugo
# add lookup for other versions based on master subject

#### Pre-Flight checklist


#### What are the relevant tickets?
https://github.com/mitodl/ocw-to-hugo/issues/273

#### What's this PR do?
In https://github.com/mitodl/ocw-data-parser/pull/74, the `other_version_parent_uids` property was added to parsed OCW JSON. This PR adds code that builds a lookup of other versions based on the keys stored in this property across the various courses being processed. Code was also added to `generateCourseHomeMarkdown` to look up other versions of a given course and generate links to them.

#### How should this be manually tested?
 - Read the readme and make sure you are set up to download courses from S3
 - Put the following in `private/courses.json`:
```
{
    "courses": [
        "8-01sc-classical-mechanics-fall-2016",
        "8-01l-physics-i-classical-mechanics-fall-2005",
        "8-01x-physics-i-classical-mechanics-with-an-experimental-focus-fall-2002"
    ]
}
```
 - Convert the courses using: `node . -i private/input -o private/output --courses private/courses.json --download --rm`
 - Inspect the contents of `private/output/8-01sc-classical-mechanics-fall-2016/content/_index.md` and ensure that the alternate versions are in the front matter:
```
other_versions:
  - >-
    [8.01L PHYSICS I: CLASSICAL
    MECHANICS](/courses/8-01l-physics-i-classical-mechanics-fall-2005) |  FALL
    2005
  - >-
    [8.01X PHYSICS I: CLASSICAL MECHANICS WITH AN EXPERIMENTAL
    FOCUS](/courses/8-01x-physics-i-classical-mechanics-with-an-experimental-focus-fall-2002)
    |  FALL 2002
```
